lib/ansible/module_utils/aws/core.py

```diff
@@ -169,12 +169,29 @@ def fail_json_aws(self, exception, msg=None):
         except AttributeError:
             response = None
 
+        boto3_version, botocore_version = self._gather_versions()
         if response is None:
-            self._module.fail_json(msg=message, exception=last_traceback)
+            self._module.fail_json(msg=message, exception=last_traceback,
+                                   boto3_version=boto3_version,
+                                   botocore_version=botocore_version)
         else:
             self._module.fail_json(msg=message, exception=last_traceback,
+                                   boto3_version=boto3_version,
+                                   botocore_version=botocore_version,
                                    **camel_dict_to_snake_dict(response))
 
+    def _gather_versions(self):
+        """Gather AWS SDK dependency versions
+
+        Returns (str, str) of boto3 and botocore versions
+        Returns ('NaN', 'NaN') if neither are installed
+        """
+        if not HAS_BOTO3:
+            return 'NaN', 'NaN'
+        import boto3
```

Comment:
We don't need the imports at this point, as we know the imports have succeeded if `HAS_BOTO3` is true.

Reply:
There's actually no direct `boto3` import in this file, everything is coming in through boto3_conn and other wrappers.

Reply:
Ah, right, hadn't realised that. 

I'm not sure any modules using AnsibleAWSModule would have got as far as module.fail_json_aws if HAS_BOTO3 is not True though. Worth checking just in case I suppose!

Reply:
Sometimes it's worth looking both ways before crossing a one-way street. My main concern is for unit-level tests that may mock out AWS libraries, but miss this helper function. In that case, I'd rather those failures just quietly miss their boto3 version info.